test/e2e_node/pod_conditions_test.go

```diff
@@ -57,13 +57,15 @@ var _ = SIGDescribe("Pod conditions managed by Kubelet", func() {
 		ginkgo.It("a pod with init containers should report all conditions set in expected order after the pod is up", runPodReadyConditionsTest(f, true, true))
 		ginkgo.It("a pod failing to mount volumes and without init containers should report scheduled and initialized conditions set", runPodFailingConditionsTest(f, false, true))
 		ginkgo.It("a pod failing to mount volumes and with init containers should report just the scheduled condition set", runPodFailingConditionsTest(f, true, true))
+		cleanupPods(f)
```

Comment:
I thought we are doing it already after each test. If it's missing here, are there other places as well?

Reply:
I was surprised as well. The "web1" pod from this test clearly showed up in another test suite where it was unexpected. I do want to keep an eye out for other places, but want to get this in first as there is evidence that it was not cleaned.

Reply:
Pods created by a given test will be cleaned up at the end of the test. This is done because the test framework deletes the namespace created by each test. However, this is done asynchronously, so there is no gurantee that the next test will start only after the resources in the last test were cleaned up.

This is by design, please see https://kubernetes.slack.com/archives/C09QZ4DQB/p1675206504013289 as discussion about it.

Reply:
@bobbypage unfortunately i just ran into one of those :(